Google.PowerShell/Compute/GceManagedInstanceGroupCmdlets.cs

```diff
@@ -454,27 +452,52 @@ protected override void ProcessRecord()
 
         /// <summary>
         /// Processes the arguments when the cmdlet receives a piped ManagedInstanceGroup.
+        /// This function will return true if we need to create a zone instance
+        /// and return false if we need to create a region.
         /// </summary>
-        private void ProcessPipedObjectArguments()
+        private bool ProcessPipedObjectArguments()
         {
-            // Only use argument from -Project if the user supplies it.
-            if (MyInvocation.BoundParameters.ContainsKey(nameof(Project))
-                && !string.IsNullOrWhiteSpace(Object.SelfLink))
+            bool regionSpecified = MyInvocation.BoundParameters.ContainsKey(nameof(Region));
+            bool projectSpecified = MyInvocation.BoundParameters.ContainsKey(nameof(Project));
+            bool zoneSpecified = MyInvocation.BoundParameters.ContainsKey(nameof(Zone));
+
+            if (regionSpecified && zoneSpecified)
+            {
+                throw UnknownParameterSetException;
```

Comment:
I believe this error will be confusing to the user. It will complain of an unknown parameter set "ByObject", when such a set clearly exists. A more specific exception message is needed.